src/part_file.cpp

```diff
@@ -310,19 +315,27 @@ namespace libtorrent
 			{
 				int const slot = i->second;
 				open_file(file::read_only, ec);
+				boost::shared_ptr<file> local_file = m_file;
```

Comment:
not trivially, this line depends on `open_file()` to actually initialize `m_file`. If this is moved, `open_file()` has to be moved as well, and I think it's in here to avoid the pessimisation of assuming we need to open the file even in cases were we don't.

Reply:
Yes, I also just realized that exporting a single file shouldn't require more than 1 piece.